File: app/src/main/java/com/star_zero/example/androidmvvm/presentation/shared/binding/TextInputLayoutBinding.java
package com.star_zero.example.androidmvvm.presentation.shared.binding;

import android.databinding.BindingAdapter;
import android.support.design.widget.TextInputLayout;

public class TextInputLayoutBinding {

    @BindingAdapter({"errorMessage"})
    public static void setErrorMessage(TextInputLayout view, int messageId) {
        if (messageId == 0) {
            view.setErrorEnabled(false);
        } else {
            view.setErrorEnabled(true);
            view.setError(view.getContext().getString(messageId));
        }
    }
}

File: app/src/main/kotlin/com/star_zero/example/androidmvvm/presentation/shared/binding/TextInputLayoutBinding.kt
package com.star_zero.example.androidmvvm.presentation.shared.binding

import android.databinding.BindingAdapter
import android.support.design.widget.TextInputLayout

object TextInputLayoutBinding {

    @JvmStatic
    @BindingAdapter("errorMessage")
    fun setErrorMessage(view: TextInputLayout, messageId: Int) {
        if (messageId == 0) {
            view.isErrorEnabled = false
        } else {
            view.isErrorEnabled = true
            view.error = view.context.getString(messageId)
        }
    }
}
